misc/fibonacci_formula.c

```diff
@@ -0,0 +1,32 @@
+/**
+ * \file
+ *
+ * \brief Implementation of [Fibonacci Sequence
+ * explicit formula solution](https://fabiandablander.com/r/Fibonacci.html)
+ *
+*/
+
+#include <math.h> /// for pow and sqrt
+#include <stdio.h>
+
+/*
+ * formula comes from quadratic formula and recurrence relation
+ * all the proof can be found in the related article in top of the file
+ */
+double fib(int n) {
+    return (1 / sqrt(5) * pow(((1 + sqrt(5)) / 2), n + 1)) -
+           (1 / sqrt(5) * pow(((1 - sqrt(5)) / 2), n + 1));
+}
+
+/* print first 10 elements of fibonacci sequence for testing */
+static void test () {
+    for(int i = 0; i <= 10; i++){
+        printf("%d. fibonacci sequence is: %.0f\n", i, fib(i));
```

Comment:
Please add self-test cases using `assert` (included in the `assert.h` library/header) so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or on our [**Discord server**](https://discord.gg/c7MnfGFGa6). 😃


__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Inefficient